ecs-agent/acs/session/session.go

```diff
@@ -262,8 +271,13 @@ func (s *session) startSessionOnce(ctx context.Context) error {
 		})
 		return err
 	}
+	acsConnectionMetric.Done(err)
```

Comment:
withGauge?

Reply:
And besides the duration metric, can we also have a counter metric for this?

Reply:
`withGauge()`is used to add any value to a given metric - useful when we want to add info to a metric outside of what's already provided (ie startTime, metric name, etc.)

Reply:
what's the purpose for the counter metric?